app/src/test/java/ch/hikemate/app/model/route/HikesViewModelTest.kt

```diff
@@ -878,6 +878,42 @@ class HikesViewModelTest {
         assertNull(hikesViewModel.selectedHike.value)
       }
 
+  @Test
+  fun `unsaveHike also unplans hike`() =
+      runTest(dispatcher) {
+        // Make sure the loaded hike is included in the saved hikes
+        coEvery { savedHikesRepo.loadSavedHikes() } returns
+            listOf(
+                SavedHike(
+                    id = singleOsmHike1[0].id,
+                    name = singleOsmHike1[0].name ?: "",
+                    date = firstJanuary2024))
+        hikesViewModel.refreshSavedHikesCache()
+
+        // Load a hike to be unsaved (load hikes from bounds so that the hike won't be removed from
+        // the loaded hikes list after being unsaved).
+        loadOsmHikes(singleOsmHike1)
+        // Check that the hike was loaded
+        assertEquals(singleOsmHike1.size, hikesViewModel.hikeFlows.value.size)
+
+        // Make sure the saved hikes repository unsaves the hike
+        coEvery { savedHikesRepo.removeSavedHike(any()) } returns Unit
+
+        // Try to unsave the loaded hike
+        var onSuccessCalled = false
+        hikesViewModel.unsaveHike(
+            hikeId = singleOsmHike1[0].id,
+            onSuccess = { onSuccessCalled = true },
+            onFailure = { fail("onFailure should not have been called") })
+
+        // The saved hikes repository should be called exactly once
+        coVerify(exactly = 1) { savedHikesRepo.removeSavedHike(any()) }
+        // The appropriate callback should be called
+        assertTrue(onSuccessCalled)
+        // The hike should now be marked as unsaved
+        assertNull(hikesViewModel.hikeFlows.value[0].value.plannedDate)
```

Comment:
Suggestion: this does not mean the hike is unsaved, it means it was unplanned, maybe you should check both and adapt comments accordingly?

Reply:
whoops, thats still the comments from when i copy-pasted this test. It now says `The hike should now be marked as unplanned (and unsaved, but that is a previous test)`

Reply:
Perfect 👌🏻